src/v/kafka/server/server.cc

```diff
@@ -200,6 +203,20 @@ ss::future<security::tls::mtls_state> get_mtls_principal_state(
       });
 }
 
+/*static*/ std::vector<bool> server::convert_api_names_to_key_bitmap(
+  const std::vector<ss::sstring>& api_names) {
+    std::vector<bool> res;
+    res.resize(max_api_key() + 1);
+    for (const ss::sstring& api_name : api_names) {
+        if (const auto api_key = api_name_to_key(api_name); api_key) {
+            res.at(*api_key) = true;
+            continue;
+        }
+        vlog(klog.warn, "Unrecognized Kafka API name: {}", api_name);
```

Comment:
Is this something that should be validated by the config:: system? I'm not sure that's even possible with today's config facility.

Reply:
I'm not sure if it should. Consider if something in the conversion function has changed in the next release and a value X that could be converted before cannot anymore. However X can be stored in the controller log already. If conversion made validation fail, such cluster will not start with the new version. So a warning with a fallback plan seems to be the best choice.

Reply:
I think it's okay for the conversion function to be tolerant (there's nothing constructive it can do other than logging), but we should do validation in the property instead.

Regarding the concern about changes between releases: validation of already-set values is intentionally done in a way that avoids refusing to start if something previously-valid becomes invalid.  So there's no need to be reluctant about adding validation.

Reply:
> we should do validation in the property instead

That may be done this way:
* `binding_base<>` gets a virtual `validate() { return true; }`
* `conversion_binding<>` tries the conversion in its overload of `validate()` and returns false if conversion had to fall back
* `property<>::validate()` iterates over all the bindings it knows and calls `validate()` on each of them

Did you mean something like that @jcsp? If so, I'll catch up with a change like that later.